plugin/aws/ec2InstanceMetadataJSON.go

```diff
@@ -15,26 +15,57 @@ type ec2InstanceMetadataJSON struct {
 	inst *ec2Instance
 }
 
-func newEC2InstanceMetadataJSON(inst *ec2Instance) *ec2InstanceMetadataJSON {
+func newEC2InstanceMetadataJSON(ctx context.Context, inst *ec2Instance) (*ec2InstanceMetadataJSON, error) {
```

Comment:
One interesting thing about these read-only files is that we can track an `mtime` for them. Probably not worth it for the metadata.json files, since we can just track the specific VM/container/pod's mtime.

Reply:
An mtime as in keep track of when we get new data that differs from the previous data? I think that would be interesting idea for later.

Reply:
History of metadata is another interesting idea in that vein.

Reply:
Or even a generalized `delta` for any file (that's probably too ambitious). I'll add a TODO in one of the metadata.json files to track this discussion later.